stdlib/public/core/StringLegacy.swift

```diff
@@ -43,7 +43,7 @@ extension String {
       String(_uninitializedCapacity: repeatedUTF8.count * count) { buffer in
         var total = 0
         for i in 0..<count {
-          let offset = i * repeatedUTF8.count
+          let offset = i &* repeatedUTF8.count
           let range = offset ..< offset + repeatedUTF8.count
```

Comment:
Similarly, this `+` also cannot overflow, and the `..<` operator can be replaced with `Range(uncheckedBounds:)`.

All of this is possible be cause we know the maximum offset (`count * repeatedUTF8.count`) did not overflow, and that `count > 0`. It is probably worth a comment saying that.

Reply:
The reason I suggested exploring the alternative of the unsafe raw buffer APIs above is that it may allow us to express a lot of these invariants with equivalent performance without writing out the unchecked operations here again, essentially reproducing those APIs but manually.